v2/spiffeid/id.go

```diff
@@ -0,0 +1,69 @@
+package spiffeid
+
+import (
+	"net/url"
+)
+
+type ID struct {
+	td   TrustDomain
+	path string
+}
+
+// String returns a SPIFFE ID given the trust domain and path components. For
+// example, String("example.org", "foo", "bar") returns
+// "spiffe://example.org/foo/bar".
+func String(td TrustDomain, path ...string) string {
+	panic("not implemented")
+}
+
+// Make makes a SPIFFE ID with the given trust domain and path.
+func Make(td TrustDomain, path ...string) ID {
+	panic("not implemented")
+}
+
+// Parse parses a SPIFFE ID URI from a string.
+func Parse(s string) (ID, error) {
+	panic("not implemented")
+}
+
+// Parse parses a SPIFFE ID URI from a URL.
+func ParseURL(u *url.URL) (ID, error) {
+	panic("not implemented")
+}
+
+// TrustDomain returns the trust domain of the SPIFFE ID.
+func (id ID) TrustDomain() TrustDomain {
+	return id.td
+}
+
+// In returns true if the SPIFFE ID is in the given trust domain.
+func (id ID) In(td TrustDomain) bool {
```

Comment:
nit: something like `MemberOf` might be clearer?

Reply:
I like it. I'll also rename `AllowIn` to be `AllowMemberOf`. That is much clearer. Good suggestion!